### DISCUSSION TITLE
Similar to DDM or passthrough?

### DISCUSSION THREAD

**Discussion by ozwin2** [hrvach/deskhop#69] [Q&A] [answered]
Hi i know it wasn't stated, but is this build similar to dynamic device mapping or pass through? (Where the connected usb devices are recognized as if they are always connected and therefore you do not get the devices disconnecting when switching)

I have ddm kvm but the polling rate is limited to 125Hz for a mouse, i very much like the idea of being able to have 1000Hz.

**Comment by kpfleming** [comment, marked as answer]
Sort of, yes. The computers don't ever see device disconnects, but the actual devices are not passed through. For example, if your keyboard is VIA or QMK compatible, you can't use those tools while the keyboard is connected through a DeskHop because the additional HID messages are not passed through. In fact the computers will only ever see one USB HID device attached (the DeskHop itself) which provides both keyboard and pointing functionality at the same time.

**Reply by ozwin2** [reply]
i understand more now, thanks for the prompt response, i take it if i used VIA to change the layout of my keyboard connected directly to my main computer, saving the layout, then when i run it through the deskhop it should all work fine? or even keybaord with dials/knobs would work fine, or would you have to have a stock keyboard layout without knobs?

having only the deskhop recognised is an okay sacrifice for the price of this build, it would at least afford me not having the raise any awkward conversations if usb devices are monitored on work computers as to why did i change usb devices 200 times today, with the deskhop there would be no usb changes.

**Reply by hrvach** [reply]
Yes, that case should work fine (I'm using VIA that way). This was actually done by design for the sake of security - only thing that is allowed to go INTO the device is the LED status message (1 byte of information) which is usually handled by the OS keyboard driver and fairly safe. Not suggesting VIA is unsafe or bad in any way, it's great, but think of a scenario where e.g. compromised computer A configures your VIA keyboard to have a key-triggered macro you then fire on computer B unknowingly.

**Reply by kpfleming** [reply]
Correct, any programming you do with VIA or QMK is stored in the keyboard itself. I've done exactly that and have my keyboard connected through a DeskHop and it works as expected.

I don't have dials or knobs on my keyboard so I don't know if those HID messages will pass through the DeskHop properly or not; if you check the issues list in this repository you can see some reports of specific keyboards not operating properly, which could be related to these 'advanced' features.

**Comment by hrvach** [comment]
I have to admit I have no idea what DDM is, but this is more of a proxy like Kevin described.  I've tried a couple of those polling rate measurement pages and they all suggest a value around 1000 Hz.

**Reply by ozwin2** [reply]
DDM is just a specific type of USB emulation, all devices are recognised at all times, but only 1 system has active control at any one time. unfortunately this is a fairly old tech now, most would just use software kvm, but some of us still cant download software to client/work computers. The ddm kvm i have only has 2 ddm usb slots (1 keyboard, 1 mouse), this is fairly annoying as i cannot have a seperate numberpad, when connecting a keyboard, numberpad and mouse it will only select 2 usb devices to passthrough. it is also limited to 125Hz polling rate.

out of interest what was the max polling rate of the device you tested? some mice go up to 8000Hz these days

**Reply by ozwin2** [reply]
i meant to ask, if you have a hub connected to the deskhop: mouse, keyboard, numberpad, dac. would these all switch or are you limited to the number of devices the deskhop can handle?

**Reply by hrvach** [reply]
I had to get a Logitech 1000 Hz mouse for this project, I had a 2$ Lenovo mouse before and I was perfectly happy with it. No idea about 8000 Hz mice, but the box asks the host to poll at 1000 Hz rate so I'm pretty sure it will not go above unmodified. 1000 Hz seems good enough for now.

I haven't tried a numpad ever, or two keyboards in one hub, but theoretically it might work. For the DAC, it will not work as this only does keyboards and mice.
